### PR TITLE
Direct PWM control of the main outputs.

### PR DESCRIPTION
**Pull request by Pedro-Roque**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary.

**Test data / coverage**
Tested with SITL, video shown below:
https://drive.google.com/open?id=1ojMMRyiiii7A6pIMROro7hTnzVVwQ9em
I can provide logs (is it possible to take them from SITL?)

**Describe problem solved by the proposed pull request**
This PR enables direct pwm control of the motors by bypassing the mixer results. A control mode flag is set when mavros ActuatorControl->group_mix = 99 (this shall be changed by something more convenient, maybe a change on the message definition to add a constant for that. This control flag is verified at the FMU level (and on PWMSim for the simulation part), and if true, then the actuator_controls from the mixer are overridden. This happens before the safety constraints, so the user inputs still go through them to make sure they are bounded and within the usual limits. 

This is a desired feature from the Automatic Control standpoint, as there is the need to have a very low-level control on the vehicle. This solution also does not interfere with any other modules or safety measures/routines. 

Minor: fixed a small leak on the FMU, where the _adc_sub was not being unsubscribed at the destructor. 

**Describe your preferred solution**
Proposed solution.

**Describe possible alternatives**
-

**Additional context**
-

### CODE REVIEW

File: src/drivers/px4fmu/fmu.cpp [hunk 1]
```diff
@@ -63,6 +63,13 @@
 #include <uORB/topics/multirotor_motor_limits.h>
 #include <uORB/topics/parameter_update.h>
 #include <uORB/topics/safety.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/vehicle_command.h>
+#include <uORB/topics/vehicle_status.h>
+
+#ifdef HRT_PPM_CHANNEL
+# include <systemlib/ppm_decode.h>
+#endif
```
Comment (bresch) [inline src/drivers/px4fmu/fmu.cpp]: Why do you need that 
```
#include <uORB/topics/vehicle_command.h>
#include <uORB/topics/vehicle_status.h>
#ifdef HRT_PPM_CHANNEL
# include <systemlib/ppm_decode.h>
#endif
``` 
?

Comment (Pedro-Roque) [reply src/drivers/px4fmu/fmu.cpp]: Hm, I don't, but maybe it was already on the tree or upon the merge. I'll remove it.

File: src/drivers/px4fmu/fmu.cpp [hunk 2]
```diff
@@ -1327,7 +1345,21 @@ PX4FMU::cycle()
 			}
 		}
 
+#endif
```
Comment (bresch) [inline src/drivers/px4fmu/fmu.cpp]: ?

File: src/drivers/px4fmu/fmu.cpp [hunk 3]
```diff
@@ -1327,7 +1345,21 @@ PX4FMU::cycle()
...
+		/* check control mode state */
+		bool updated = false;
```
Comment (bresch) [inline src/drivers/px4fmu/fmu.cpp]: Already defined a few lines above

**Comment by Pedro-Roque** [comment]
I think I did something wrong on the merge. Give me a few moments

**Comment by bresch** [comment]
Please rebase on top of master instead of merging master into you branch

**Comment by Pedro-Roque** [comment]
> Please rebase on top of master instead of merging master into you branch

Yep, should have done that instead. Will take care of it in a second.

**Comment by Pedro-Roque** [comment]
@bresch  should I open another PR?

**Comment by Pedro-Roque** [comment]
PR to master here: https://github.com/PX4/Firmware/pull/10863/

**Pull request closed without merging**